test/node-api/test_cannot_run_js/test.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+// Test that `napi_call_function()` returns `napi_cannot_run_js` in experimental
+// mode and `napi_pending_exception` otherwise. This test calls the add-on's
+// `createRef()` method, which creates a strong reference to a JS function. When
+// the process exits, it calls all reference finalizers. The finalizer for the
+// strong reference created herein will attempt to call `napi_call_function()`
+// and will abort the process if the API doesn't return the correct status.
+
+const { buildType } = require('../../common');
+const addon_v8 = require(`./build/${buildType}/test_pending_exception`);
+const addon_new = require(`./build/${buildType}/test_cannot_run_js`);
+
+function runTests(addon, isVersion8) {
+  addon.createRef(runTests);
```

Comment:
Since this function isn't expected to be called, it would be better to assert that it wouldn't be called.

```suggestion
  addon.createRef(common.mustNotCall());
```